### PR TITLE
convert FastRarefaction to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
this gives identical answers to the Fortran

File: Exec/mhd_tests/FastRarefaction/problem_initialize_mhd_data.H [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#if AMREX_SPACEDIM == 3
+    z = problo[2] + dx[2] * (static_cast<Real>(k) + 0.5_rt);
+#endif
+
+
+    if (problem::idir == 1) {
+        // initialize along the x-direction
+
+        if (ibdir == 0) {
+            if (x <= problem::split[0]+1) {
```
Comment (maxpkatz) [inline Exec/mhd_tests/FastRarefaction/problem_initialize_mhd_data.H:31]: Can you explain where the +1 comes from here?

Comment (zingale) [reply Exec/mhd_tests/FastRarefaction/problem_initialize_mhd_data.H:31]: oh, I think that is a copy-paste from a different problem
I'll fix it

Comment (zingale) [reply Exec/mhd_tests/FastRarefaction/problem_initialize_mhd_data.H:31]: oh wait... indeed that was in the mag intiialization originally, but not in the hydro state.  I am not sure why that is there.

Comment (zingale) [reply Exec/mhd_tests/FastRarefaction/problem_initialize_mhd_data.H:31]: it looks like it is only in the problem direction that we do that.

Comment (maxpkatz) [reply Exec/mhd_tests/FastRarefaction/problem_initialize_mhd_data.H:31]: yeah, DaiWoodward did that too, that's weird but whatever.

**Comment by maxpkatz** [comment]
Looks like you also need to rm the prob_mhd_nd.f90

**Comment by zingale** [comment]
done

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**